### PR TITLE
[wip] Rename onmessagequeueempty to flush in java destinations

### PR DESCRIPTION
**Pull request by lbudai**
This should merge only after #2063 is merged.

### CODE REVIEW

**Comment by furiel** [comment]
The patch itself looks good to me, but it feels incomplete. In a previous version of #2063 there was an additional condition before flush in logthrdestdrv.c: batch size must not be empty. I think we should add this to make flush api complete: no reason to flush on an empty batch. 

Though that introduces another problem. As currently the language binding inserts return with boolean (we are facing #1761 again) we effectively disable batch insert: either a message is dropped or accepted, no way to mark as queued. With onEmpty behaviour, users could simulate batch sending by accepting each message, save them inside the binding, then flush with the onEmpty notification. Now there is no convenient way to flush, except counting inside the binding + timeout.

In short, language bindings expect onEmpty api until #1761.

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by lbudai** [comment]
@furiel: Thanks for the note. My plan is to discuss this topic on next OSE handover.

**Comment by kira-syslogng** [comment]
Build FAILURE

File: modules/java/src/main/java/org/syslog_ng/LogDestination.java [hunk 1]
```diff
@@ -56,8 +56,8 @@ public int getSeqNum() {
 
 	private native int getSeqNum(long ptr);
 
-	protected void onMessageQueueEmpty() {
-		return;
+	protected boolean flush() {
```
Comment (Kokan) [inline modules/java/src/main/java/org/syslog_ng/LogDestination.java]: I understand that we want to change the interface here, and without deprecating the old one. (I think we could even get away with it.)
But if we accept this change, does this change enough ?
Are we going to feel that the java interface is good enough for at least mid term ?
I think we should hurt the user once, and not continuously.

**Comment by lbudai** [comment]
@kokan: see my last comment ;)
my aim with this patch set was to point out some problems, including this one.
after batching has been merged, we need to figure out the next steps.
